src/sdk/namespace_fs.js

```diff
@@ -2054,6 +2054,17 @@ class NamespaceFS {
         dbg.log0('namespace_fs.restore_object:', params);
 
         await this._throw_if_storage_class_not_supported(s3_utils.STORAGE_CLASS_GLACIER);
+        if (params.days > config.NSFS_GLACIER_RESTORE_MAX_DAYS) {
+            if (config.NSFS_GLACIER_RESTORE_MAX_DAYS_BEHAVIOUR === 'DENY') {
+                throw new S3Error({
+                    ...S3Error.BadRequest,
+                    detail: `restore days cannot be more than ${config.NSFS_GLACIER_RESTORE_MAX_DAYS}`},
+                );
+            }
+
+            dbg.warn(`restore_object days exceed max configured days (${config.NSFS_GLACIER_RESTORE_MAX_DAYS}) - truncating`);
+            params.days = config.NSFS_GLACIER_RESTORE_MAX_DAYS;
+        }
```

Comment:
Logic looks good now, but now that I see this I would like to generalize the check for S3, not just nsfs, as this is a pure protocol check that is not specific to nsfs. So I suggest to rename the configs to have S3_ prefix instead of NSFS_ and refactor. Notice I changed below to return S3Error.InvalidArgument perhaps it fits better than BadRequest?

In s3_post_object_restore.js to extract [this code block](https://github.com/noobaa/noobaa-core/blob/5914557f0ee3e932093d22e92ec96be068d3618b/src/endpoint/s3/ops/s3_post_object_restore.js#L14-L23) and just call a utility function:
```js
const days = s3_utils.parse_restore_request_days(req);
...
```

in s3_utils:
```js
function parse_restore_request_days(req) {
    if (!req.body?.RestoreRequest?.Days?.[0]) {
        dbg.warn('parse_restore_request_days: missing Days in body');
        throw new S3Error(S3Error.MalformedXML);
    }

    const days = s3_utils.parse_decimal_int(req.body.RestoreRequest.Days[0]);
    if (days < 1) {
        dbg.warn('parse_restore_request_days: days cannot be less than 1');
        throw new S3Error(S3Error.InvalidArgument);
    }

    if (days > config.S3_RESTORE_REQUEST_MAX_DAYS) {
        if (config.S3_RESTORE_REQUEST_MAX_DAYS_BEHAVIOR === 'DENY') {
            throw new S3Error({
                ...S3Error.InvalidArgument,
                detail: `Restore request days ${days} is above max ${config.S3_RESTORE_REQUEST_MAX_DAYS}`},
            );
        }

        dbg.log0(`Restore request days ${days} is above max ${config.S3_RESTORE_REQUEST_MAX_DAYS} - truncating`);
        return config.S3_RESTORE_REQUEST_MAX_DAYS;
    }

    return days;
}
```

Reply:
Actually I did initially consider to place this at the more generic location but then figured that probably we might not need this config everywhere. For example any cloud backed storages. WDYT?

Reply:
You are right that for buckets with unlimited cloud archive tier it might be preferrable to have the limit higher, but I still don't think that functionally it makes sense to have no limit - notice that S3 client has no control over premature expiry so once you set it to 100 years you are kind of stuck.

So I would say this - eventually this might make sense as a per-bucket option, but for now I am ok with this being a system wide config, and I would apply it to all types of buckets.